src/main/java/org/broadinstitute/hellbender/utils/IndexUtils.java

```diff
@@ -48,43 +70,92 @@ public static Index loadTribbleIndex(final File featureFile) {
      */
     public static Index loadTabixIndex(final File featureFile) {
         Utils.nonNull(featureFile);
+
+        final File indexFile = getTabixIndexFile(featureFile);
+        if ( indexFile != null ) {
+            logger.debug("Loading tabix index from disk for file " + featureFile);
+            final Index index = IndexFactory.loadIndex(indexFile.getAbsolutePath());
+            checkIndexVersion(featureFile, indexFile, index);
+            return index;
+        }
+
+        return null;
+    }
+
+    /**
+     * Try to get the {@link File} representing the tabix index on disk
+     * Does not perform any checks for validity.
+     * @return a {@link File}, or null if we're unable to find it.
+     */
+    public static File getTabixIndexFile(final File featureFile) {
+        Utils.nonNull(featureFile);
         try {
             final String path = featureFile.getAbsolutePath();
             final boolean isTabix = new AbstractFeatureReader.ComponentMethods().isTabix(path, null);
             if (! isTabix){
                 return null;
             }
             final String indexPath = ParsingUtils.appendToPath(path, TabixUtils.STANDARD_INDEX_EXTENSION);
-            logger.debug("Loading tabix index from disk for file " + featureFile);
-            final Index index = IndexFactory.loadIndex(indexPath);
-            final File indexFile = new File(indexPath);
-            checkIndexVersionAndModificationTime(featureFile, indexFile, index);
-            return index;
+            logger.debug("Getting tabix file on disk for feature file: " + featureFile);
+            return new File(indexPath);
         } catch (final IOException | RuntimeException e) {
             return null;
         }
     }
 
     /**
      * @throws UserException the index is not the current version.
-     * Prints a warning if the index is not up-to date (ie older than the feature file)
      */
-    public static void checkIndexVersionAndModificationTime(final File featureFile, final File indexFile, final Index index) {
+    public static void checkIndexVersion(final File featureFile, final File indexFile, final Index index) {
         Utils.nonNull(featureFile, "featureFile");
         Utils.nonNull(indexFile, "indexFile");
         Utils.nonNull(index, "index");
         if (! index.isCurrentVersion()) {
             // we've loaded an old version of the index, we want to remove it
-            throw new UserException("Index file " + indexFile + " is out of date (old version). Use " + IndexFeatureFile.class.getSimpleName() +  " to make an index.");
-        } else if (indexFile.lastModified() < featureFile.lastModified()) {
-            logger.warn("Index file " + indexFile + " is out of date (index older than input file). Use " + IndexFeatureFile.class.getSimpleName() +  " to make a new index.");
+            throw new UserException.OutOfDateIndex(indexFile.getAbsolutePath(), IndexFeatureFile.class.getSimpleName());
```

Comment:
I would throw a message that says it's the file version that's the problem. If it's trivial, output what he version should be / is.

Reply:
Fixed with a new exception type.

I don't see a quick/easy way to get the index version in these interfaces...